### PR TITLE
Add testing mode

### PR DESCRIPTION
**Pull request by hynek**
ref #59

Docs: https://py-stamina--73.org.readthedocs.build/en/73/testing.html

@wlach @tino for your consideration!

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+This is a great approach when you're only using our decorator-based API.
+
+
+## Limiting Retries
+
+:::{versionadded} 24.3.0
+:::
+
+When you need more control, you will use the iterator-based APIs around {func}`stamina.retry_context`.
```
Comment (wlach) [inline docs/testing.md:27]:
```suggestion
When you need more control, you can use the iterator-based APIs around {func}`stamina.retry_context`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ You can find our backwards-compatibility policy [here](https://github.com/hynek/
 - `stamina.Attempt` now has a `next_wait` attribute that contains the time the *next* backoff will wait, if the *current* attempt fails (sans jitter).
   [#72](https://github.com/hynek/stamina/pull/72)
 
+- It is now possible to switch *stamina* into a testing mode using `stamina.set_testing()`.
+  It disables backoffs and caps the number of retries.
```
Comment (wlach) [inline CHANGELOG.md:27]: You probably want to link to the PR here?

File: tests/typing/api.py [hunk 3]
```diff
@@ -94,6 +96,14 @@ def exc_tune_waiting_timedelta() -> None: ...
 
 set_active(False)
 
+set_testing(True)
+set_testing(True, attempts=2)
+
+if is_testing() is False:
```
Comment (wlach) [inline tests/typing/api.py:102]: Just curious, is there a reason why you're testing via `is False` here rather than `not is_testing()`?

**Review by wlach — COMMENTED** [review]
❤️ This is great, I think this API will be very usable. I didn't test the code

File: CHANGELOG.md (hunk 2 above)
Comment (hynek) [reply CHANGELOG.md:27]: yes, thanks!

**Pull request merged**